guide/canvas/README.md

```diff
@@ -0,0 +1,334 @@
+
+	ctx.beginPath();
+	ctx.arc(125, 125, 100, 0, Math.PI * 2, true);
+	ctx.closePath();
+	ctx.clip();
+
+	const buffer = await snekfetch.get(member.user.displayAvatarURL).then(r => r.body);
+	const avatar = await Canvas.loadImage(buffer);
+	ctx.drawImage(avatar, 25, 25, 200, 200);
+
+	const attachment = new Discord.Attachment(canvas.toBuffer(), 'image.png');
+
+	channel.send(`Welcome to the server, ${member}`, attachment);
+});
+```
+
+If followed successfully, the image should now look like this: ![Image](/assets/img/3rLGb1s.png)
+
+If you get an error like `Fontconfig error: Cannot load default config file`, it means you do not have any fonts installed on your system. Run the following command to install it: `sudo apt-get install fontconfig`. 
+
+Maybe you have noticed, or considered, where if a member's username is too long, then the output won't be quite nice. This is because the text overflows out of the canvas, and you don't have any measures in place for that. Let's take care of this issue, shall we?
+
+```js
+const Discord = require('discord.js');
+const Canvas = require('canvas');
+const snekfetch = require('snekfetch');
+
+// Pass the entire canvas because you will want to compare its width as well while being able to grab its context
+const applyText = (canvas, username) => {
+	const ctx = canvas.getContext('2d');
+
+	// Declare a base size of the font
+	let size = 70;
+	do {
+		// Assign the font to the context and decrement it so it can be measured again
+		ctx.font = `${size -= 10}px sans-serif`;
+		// Compare pixel width of the text to the canvas minus the approximate avatar size
+	} while (ctx.measureText(username).width > canvas.width - 300);
+
+	// Return the result to use in the actual canvas
+	return ctx.font;
+};
+
+client.on('guildMemberAdd', async member => {
+	const channel = member.guild.channels.find(ch => ch.name === 'member-log');
+	if (!channel) return;
+
+	const canvas = Canvas.createCanvas(700, 250);
+	const ctx = canvas.getContext('2d');
+
+	const background = await Canvas.loadImage('./wallpaper.jpg');
+	ctx.drawImage(background, 0, 0, canvas.width, canvas.height);
+
+	ctx.strokeStyle = '#FF0000';
+	ctx.strokeRect(0, 0, canvas.width, canvas.height);
+
+	// Assign the decided font to the canvas
+	ctx.font = applyText(canvas, member.displayName);
+	ctx.fillStyle = '#FFFFFF';
+	ctx.fillText(member.displayName, canvas.width / 2.5, canvas.height / 1.8);
+
+	ctx.beginPath();
+	ctx.arc(125, 125, 100, 0, Math.PI * 2, true);
+	ctx.closePath();
+	ctx.clip();
+
+	const buffer = await snekfetch.get(member.user.displayAvatarURL).then(r => r.body);
+	const avatar = await Canvas.loadImage(buffer);
+	ctx.drawImage(avatar, 25, 25, 200, 200);
+
+	const attachment = new Discord.Attachment(canvas.toBuffer(), 'image.png');
+
+	channel.send(`Welcome to the server, ${member}`, attachment);
+});
+```
+
+Before adjustment: ![Image](/assets/img/NKw7P2q.png)<br />
+After adjustment: ![Image](/assets/img/Ja4Ywf4.png)
```

Comment:
Missing the "Resulting code" bit at the bottom like all the other pages. A finishing statement would be nice, but not required.

Reply:
Also, no need for a `<br />`; just add a blank line between the two pieces of text.